rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,141 @@
+        for _log in all_logs:
+            if _log.topics[0] in WITHDRAWAL_BONDED:
+                self._decode_withdrawal_bonded(context=DecoderContext(
+                    tx_log=_log,
+                    transaction=transaction,
+                    action_items=[],
+                    all_logs=all_logs,
+                    decoded_events=decoded_events,
+                ))
+        return decoded_events
+
+    def _decode_withdrawal_bonded(self, context: DecoderContext) -> DecodingOutput:
+        recipient = hex_or_bytes_to_address(context.transaction.input_data[4:36])
+        if not self.base.is_tracked(recipient):
+            return DEFAULT_DECODING_OUTPUT
+
+        bridge = self.bridges.get(context.tx_log.address)
+        if bridge is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        if bridge.asset == A_ETH:
+            normalized_amount = token_normalized_value_decimals(amount_raw, 18)
+        else:
+            normalized_amount = token_normalized_value(amount_raw, bridge.asset.resolve_to_evm_token())  # noqa: E501
+
+        for event in context.decoded_events:
+            if (
+                event.address in [bridge.amm_wrapper, bridge.address] and
+                event.asset == bridge.asset and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} via hop protocol'  # noqa: E501
+                event.extra_data = {'sent_amount': str(normalized_amount)}
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_receive(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == WITHDRAWAL_BONDED:
+            self._decode_withdrawal_bonded(context=context)
+
+        if context.tx_log.topics[0] == TRANSFER_FROM_L1_COMPLETED:
+            recipient = hex_or_bytes_to_address(context.tx_log.topics[1])
+            if not self.base.is_tracked(recipient):
+                return DEFAULT_DECODING_OUTPUT
+
+            bridge = self.bridges.get(context.tx_log.address)
+            if bridge is None:
+                return DEFAULT_DECODING_OUTPUT
+            amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+            heth_amount = token_normalized_value_decimals(amount_raw, 18)
+            logger.info(context.decoded_events)
+
+            for event in context.decoded_events:
+                if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and recipient == event.location_label and event.asset == bridge.asset:  # noqa: E501
```

Comment:
same, shouldn't you also check on the amount? and break when you find it

Reply:
same here, added break, not matching the amount, cause this amount still has the fees, but the transfer decoded event, doesn't.